solr/core/src/java/org/apache/solr/handler/admin/SegmentsInfoRequestHandler.java

```diff
@@ -434,4 +435,9 @@ public String getDescription() {
   public Category getCategory() {
     return Category.ADMIN;
   }
+
+  @Override
+  public Name getPermissionName(AuthorizationContext request) {
+    return Name.CONFIG_READ_PERM;
```

Comment:
I had to change the predefined perm `METRICS_READ_PERM` to `'*' or null` so that it will work both for node-level handler and for collection-level handler such as <coll>/admin/segments. This authz system is quite trappy!

So now the segments handler is guarded by metrics-read. I also tested the ping handler, and it is guarded by the new 'health' permission.